src/algebra/ordered_monoid_lemmas.lean

```diff
@@ -284,78 +595,39 @@ lemma lt_mul_of_lt_of_one_lt' (hbc : b < c) (ha : 1 < a) :
   b < c * a :=
 lt_mul_of_lt_of_one_le' hbc ha.le
 
-end covariant
+end covariant_left
 
-section contravariant
+section covariant_right
 variable [covariant_class α α (function.swap (*)) (≤)]
 
 @[to_additive add_pos_of_nonneg_of_pos]
 lemma one_lt_mul_of_le_of_lt' (ha : 1 ≤ a) (hb : 1 < b) : 1 < a * b :=
 lt_of_lt_of_le hb $ le_mul_of_one_le_left' ha
 
 @[to_additive]
-lemma lt_mul_of_one_le_of_lt' (ha : 1 ≤ a) (hbc : b < c) : b < a * c :=
+lemma lt_mul_of_one_le_of_lt (ha : 1 ≤ a) (hbc : b < c) : b < a * c :=
 hbc.trans_le $ le_mul_of_one_le_left' ha
 
 @[to_additive]
-lemma lt_mul_of_one_lt_of_lt' (ha : 1 < a) (hbc : b < c) : b < a * c :=
-lt_mul_of_one_le_of_lt' ha.le hbc
-
-end contravariant
+lemma lt_mul_of_one_lt_of_lt (ha : 1 < a) (hbc : b < c) : b < a * c :=
+lt_mul_of_one_le_of_lt ha.le hbc
 
-variables [covariant_class α α (*) (≤)] [covariant_class α α (function.swap (*)) (≤)]
+end covariant_right
 
-@[to_additive]
-lemma le_mul_of_one_le_of_le (ha : 1 ≤ a) (hbc : b ≤ c) : b ≤ a * c :=
-one_mul b ▸ mul_le_mul' ha hbc
+-- up to here
```

Comment:
I changed this.  Since the `left` and `right` assumptions are now split, it is a little awkward: there might be the need to choose whether increasing `has_le` to `preorder` should be done before or after switching to the appropriate `left/right` assumption.